Wrap beatmap listing filters and match web spacing

Joehuu opened the pull request:
- Resolves https://github.com/ppy/osu/issues/27532

Had this change but thought it was unimportant at the time (a month ago) and the design is *[changing again](https://www.figma.com/file/bONtBvDRWNfOuiYpueD2ms/Web%2FSearch-Result-(with-Beatmap-Listing)?type=design&node-id=46-248&mode=design&t=VyKVQc74eqhNvcJf-0)*. My intention at the time was to fix the display for high ui scales but didn't account for other languages.

| Before | After | Web |
| --- | --- | --- |
| <img width="906" alt="Screenshot 2024-03-08 at 12 09 12 PM" src="https://github.com/ppy/osu/assets/35318437/cdb67f5e-6a38-4d30-aa53-544f3dacef38"> | <img width="876" alt="Screenshot 2024-03-08 at 12 07 18 PM" src="https://github.com/ppy/osu/assets/35318437/d2e39b7a-c5f7-4520-b18c-95ccce0be246"> | <img width="927" alt="Screenshot 2024-03-08 at 12 14 45 PM" src="https://github.com/ppy/osu/assets/35318437/29a850b1-8a40-481a-ac4f-b0d67a402636"> |

Joehuu committed:
Wrap beatmap listing filters and match web spacing

bdach commented:
> the design is _[changing again](https://www.figma.com/file/bONtBvDRWNfOuiYpueD2ms/Web%2FSearch-Result-(with-Beatmap-Listing)?type=design&node-id=46-248&mode=design&t=VyKVQc74eqhNvcJf-0)_

I've seen this months ago already. And it probably hasn't been reviewed by anyone yet, so I wouldn't worry about it.

bdach committed:
Wrap beatmap listing filter names too

While we're here fixing...

Addresses
https://github.com/ppy/osu/discussions/15452#discussioncomment-2734237.

bdach committed:
Merge branch 'master' into wrap-beatmap-listing-filters

bdach commented:
Tacked on https://github.com/ppy/osu/pull/27541/commits/e5e7c8f26841654a6819f3c88c40a347184c086b while we're here. Seems fine otherwise.
